File: read_headers.py
import openpyxl
import json

wb = openpyxl.load_workbook(
    'templates/products-and-offers-en_GB-20260302191642.xlsx',
    read_only=True,
    data_only=True
)

ws = wb.worksheets[0]
print(f"Sheet name: {ws.title}")

rows_checked = 0
for i, row in enumerate(ws.iter_rows(max_row=10, values_only=True), start=1):
    non_empty = [v for v in row if v is not None and v != '']
    if len(non_empty) >= 2:
        print(f"ROW {i}: {json.dumps(list(row), default=str)}")
        rows_checked += 1
        if rows_checked >= 3:
            break

wb.close()

File: read_headers.js
const ExcelJS = require('./node_modules/exceljs/lib/exceljs.nodejs.js');
const fs = require('fs');
const wb = new ExcelJS.Workbook();
wb.xlsx.readFile('./templates/products-and-offers-en_GB-20260302191642.xlsx').then(() => {
  const ws = wb.worksheets[0];
  const lines = [];
  lines.push('Sheet name: ' + ws.name);
  lines.push('Row count: ' + ws.rowCount);
  lines.push('Col count: ' + ws.columnCount);
  for (let n = 1; n <= 5; n++) {
    const row = ws.getRow(n);
    const vals = row.values; // 1-indexed array
    if (!vals) continue;
    const nonEmpty = vals.filter(v => v !== null && v !== undefined && v !== '');
    if (nonEmpty.length >= 2) {
      lines.push('ROW ' + n + ' ' + JSON.stringify(vals));
    }
  }
  const out = lines.join('\n') + '\n';
  fs.writeFileSync('./headers_output.txt', out, 'utf8');
  process.stdout.write(out);
}).catch(e => {
  const msg = 'ERROR: ' + e.message + '\n';
  fs.writeFileSync('./headers_output.txt', msg, 'utf8');
  process.stderr.write(msg);
  process.exit(1);
});
